cmd/utils/cmd.go

```diff
@@ -185,32 +190,87 @@ func ExportChain(blockchain *core.BlockChain, fn string) error {
 		defer writer.(*gzip.Writer).Close()
 	}
 
-	if err := blockchain.Export(writer); err != nil {
+	var z *zip.Writer
+	if strings.HasSuffix(fn, ".zip") {
+		z = zip.NewWriter(fh)
```

Comment:
I think it would be cleaner and less error prone to have `defer z.Close()` here. I'm unsure if we need specific error handling for it, if something fails, it will probably not be the close.